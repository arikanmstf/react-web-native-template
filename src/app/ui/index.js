export { default as Text } from './Text/Web';
export { default as Button } from './Button/Web';
export { default as View } from './View/Web';
export { default as Image } from './Image/Web';
