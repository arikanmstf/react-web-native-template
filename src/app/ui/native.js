export { default as Text } from './Text/Native';
export { default as Button } from './Button/Native';
export { default as View } from './View/Native';
export { default as Image } from './Image/Native';
