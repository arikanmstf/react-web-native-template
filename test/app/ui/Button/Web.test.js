import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'ui';

describe('src/app/ui/Button/Web.js', () => {
  it('Render: props: { text }', () => {
    const props = {
      text: 'dummyText',
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { primary }', () => {
    const props = {
      text: 'dummyText',
      primary: true,
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { raised }', () => {
    const props = {
      text: 'dummyText',
      raised: true,
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { onPress }', () => {
    const props = {
      text: 'dummyText',
      onPress: jest.fn(() => 'returnValue'),
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
