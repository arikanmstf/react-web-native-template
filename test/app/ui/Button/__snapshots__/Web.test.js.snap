// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/ui/Button/Web.js Render: props: { onPress } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onPress={[MockFunction]}
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": undefined,
      "onPress": [MockFunction],
      "variant": undefined,
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": undefined,
        "onPress": [MockFunction],
        "variant": undefined,
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`src/app/ui/Button/Web.js Render: props: { primary } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    primary={true}
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": "primary",
      "variant": undefined,
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": "primary",
        "variant": undefined,
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`src/app/ui/Button/Web.js Render: props: { raised } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    raised={true}
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": undefined,
      "variant": "contained",
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": undefined,
        "variant": "contained",
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`src/app/ui/Button/Web.js Render: props: { text } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": undefined,
      "variant": undefined,
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": undefined,
        "variant": undefined,
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
